### ISSUE TITLE
Verification of recent changes in DDE4 Simulator

### ISSUE THREAD

**Issue by cfry** [cfry/dde#90]
DDE4 simulator may not behave as well as DDE3.
This task is to verify that DDE4's simulator is up
to DDE3 and to fix incompatibilities.

**Comment by JamesNewton** [comment]
I'm still having intermittent problems with the DDE4 simulator. For example, note the J2 slider is at 25 degrees, the job is running, but the simulated arm is still at home:
![image](https://github.com/cfry/dde/assets/419392/4b1ffe5f-0000-4887-9a3d-aba6344ec421)
here are the errors I get:
![image](https://github.com/cfry/dde/assets/419392/67f18ea2-060c-4736-b44c-a6b913d8df8b)

I did wait for the simulated robot to show up before I continued. Note this is behind a certain VPN. On another PC, not using a VPN, the first two sets of errors shows up, but the 3rd one, Cannot read properties of undefined (reading 'rotation') does NOT show up, and the simulator seems to work. So apparently, again, VPN problems. Not sure why...
